docs/source/basics/defining_environment.md

```diff
@@ -0,0 +1,239 @@
+  - codecov
+variables:
+  VAR1: valueA
+  VAR2: valueB
+```
+</details>
+<br />
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+env CONDA_DIR /opt/conda
+env PATH=$CONDA_DIR/bin:$PATH
+run apt-get update --yes && \
+    apt-get install --yes curl && \
+    curl --remote-name https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
+    mkdir /root/.conda && \
+    # docs for -b and -p flags: https://docs.anaconda.com/anaconda/install/silent-mode/#linux-macos
+    bash Miniconda3-latest-Linux-x86_64.sh -b -p /opt/conda && \
+    rm -f Miniconda3-latest-Linux-x86_64.sh && \
+    conda init bash
+copy environment.yml /opt/latch/environment.yml
+run conda env create --file /opt/latch/environment.yml --name workflow
+shell ["conda", "run", "--name", "workflow", "/bin/bash", "-c"]
+run pip install --upgrade latch
+```
+</details>
+<br />
+
+### R Packages
+A local R environment will be created using an `environment.R` script. This script should include all the R packages needed to run the relevant R scripts in a workflow. R 4.0 is the default version.
+
+<details>
+<summary>Example File</summary>
+
+```R
+install.packages("RCurl")
+install.packages("BiocManager")
+BiocManager::install("S4Vectors")
+```
+</details>
+<br />
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+run apt-get update --yes && \
+    apt-get install --yes software-properties-common && \
+    add-apt-repository "deb http://cloud.r-project.org/bin/linux/debian buster-cran40/" && \
+    apt-get install --yes r-base r-base-dev libxml2-dev libcurl4-openssl-dev libssl-dev wget
+copy environment.R /opt/latch/environment.R
+run Rscript /opt/latch/environment.R
+```
+</details>
+<br />
+
+### System (Debian) Packages
+Requirements found in `system-requirements.txt` are installed system-wide using apt.
+
+<details>
+<summary>Example File</summary>
+
+```
+autoconf
+samtools
+```
+</details>
+<br />
+
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+copy system-requirements.txt /opt/latch/system-requirements.txt
+run apt-get update --yes && xargs apt-get install --yes </opt/latch/system-requirements.txt
+```
+</details>
+<br />
+
+### Environment Variables
+Environment variables found in an `environment` file will be used to set environment variables inside the workflow environment.
+
+<details>
+<summary>Example File</summary>
+
+```
+BOWTIE2_INDEXES=reference
+PATH="$PATH:/root/bowtie2"
+```
+</details>
+<br />
+
+<details>
+<summary>Docker commands</summary>
+
+```Dockerfile
+env {line1}
+env {line2}
+...
+```
+</details>
+<br />
+
+---
+<br>
+
+To save the generated Dockerfile in the workflow directory, run `latch dockerfile {path_to_workflow_directory}`. The generated Dockefile will be used during subsequent `latch register` and `latch develop` commands to build the workflow environment, and it can be user modified. To go back to the autogenerated Dockerfile, delete the Dockerfile in the workflow directory.
```

Comment:
- rephrase in a more factual tone (rather than a how-to tone)
- rewrite the command line parameter to be in canonical Unix notation (except Unix also wants you to write the parameter in all-caps which I aint doin)
- rewrite the second sentence in active voice instead of passive
- added more details
- added suggested use-cases
- added a second suggestion-beg since its relevant here
```suggestion
The auto-generated Dockerfile can be saved to the workflow root using `latch dockerfile <path to workflow root>`. Subsequent `latch register` and `latch develop` commands will use the saved version. This also disables automatic generation so no dependency files will be used and changes in these files will not have any effect. Removing the Dockerfile will re-enable automatic generation.

This can be used to switch to a more complicated handwritten Dockerfile or to debug any issues with auto-generation.

If you use ejection because auto-generation does not cover your use case, please [open a suggestion on GitHub.](https://github.com/latchbio/latch/issues)
```